packages/context/src/resolver.ts

```diff
@@ -230,10 +234,10 @@ export function resolveInjectedArguments(
         return extraArgs[nonInjectedIndex++];
       } else {
         const name = getTargetName(target, method, ix);
-        throw new Error(
-          `Cannot resolve injected arguments for ${name}: ` +
-            `The arguments[${ix}] is not decorated for dependency injection, ` +
-            `but a value is not supplied`,
+        throw new ResolutionError(
+          `The value is not supplied or injected for argument '${name}'.` +
+            ` Please check if @inject is missing.`,
```

Comment:
Personally, I find the new messaging as making it more difficult to understand what's wrong. However, I do like the hint on how to fix the problem.

I am proposing the following:

```suggestion
        throw new ResolutionError(
          `The argument '${name}' is not decorated for dependency injection ` +
            'and no value was supplied by the caller. Did you forget to call '
            '@inject() decorator to configure dependency injection?'
```